pkg/cmd/server/start/start_allinone.go

```diff
@@ -209,6 +218,33 @@ func (o *AllInOneOptions) Complete() error {
 	o.NodeArgs.NodeName = strings.ToLower(o.NodeArgs.NodeName)
 	o.NodeArgs.MasterCertDir = o.MasterOptions.MasterArgs.ConfigDir.Value()
 
+	// For backward compatibility, enabling node DNS continues to start the master DNS, but the
+	// container DNS server will be the node's. However, if the user has provided an override DNSAddr,
+	// we need to honor the value if the port is not 53 and we do that by disabling node DNS.
+	if !o.IsRunFromConfig() && o.NodeArgs.Components.Enabled(ComponentDNS) {
+		dnsAddr := &o.MasterOptions.MasterArgs.DNSBindAddr
+
+		// if the user set the DNS port to anything but 53, disable node DNS
+		if dnsAddr.Provided {
+			if dnsAddr.Port == 53 {
+				// continue to run the master DNS on port 8053
+				dnsAddr.Port = 8053
```

Comment:
this looks like it is changing a value explicitly set by the user. am I misreading that?